docs(python): Update GPU engine installation instructions to remove `--extra-index-url` from CUDA 12 packages (#20381)
pola-rs/polars: docs/source/user-guide/gpu-support.md

=== ":fontawesome-brands-python: Python"

```bash
pip install polars[gpu]
```

!!! note Installation on a CUDA 11 system

>     If you have CUDA 11, the installation line is slightly more complicated: the relevant GPU package must be requested by hand using the NVIDIA package index.

Reviewer: Maybe this is simpler?
```suggestion
    If you have CUDA 11, the installation line also needs the NVIDIA package index to get the CUDA 11 package.
```
